tree/treeplayer/src/TSimpleAnalysis.cxx

```diff
@@ -0,0 +1,384 @@
+
+   // Check if there are histograms with the same name
+   if (!check.second)
+      return "Duplicate histogram name";
+   return "";
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Constructor for the case of command line parsing arguments. It sets the members
+/// of the object.
+///
+/// \param[in] output name of the output file
+/// \param[in] inputFiles name of the input .root files
+/// \param[in] expressions what is showed in the histograms
+/// \param[in] treeName name of the tree
+/// \throws std::runtime_error in case of ill-formed expressions
+
+TSimpleAnalysis::TSimpleAnalysis(const std::string& output,
+                                 const std::vector<std::string>& inputFiles,
+                                 const std::vector<std::string>& expressions,
+                                 const std::string& treeName = ""):
+   fInputFiles(inputFiles), fOutputFile(output), fTreeName(treeName)
+{
+   for (const std::string& expr: expressions) {
+      std::string errMessage = HandleExpressionConfig(expr);
+      if (!errMessage.empty())
+         throw std::runtime_error(errMessage + " in " +  expr);
+   }
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Extract the name of the tree from the first input file when the tree name
+/// isn't in the configuration file. Returns the name of the tree.
+
+std::string TSimpleAnalysis::ExtractTreeName()
+{
+   std::string treeName = "";
+   TFile inputFile (fInputFiles[0].c_str());
+
+   // Loop over all the keys inside the first input file
+   for (TObject* keyAsObj : *inputFile.GetListOfKeys()) {
+      TKey* key = dynamic_cast<TKey*>(keyAsObj);
+      TClass* clObj = TClass::GetClass(key->GetClassName());
+      if (!clObj)
+         continue;
+      // If the key is releted to and object that inherits from TTree::Class we
+      // set treeName with the name of this key if treeName is empty, otherwise
+      // error occours
+      if (clObj->InheritsFrom(TTree::Class())) {
+         if (treeName.empty())
+            treeName = key->GetName();
+         else {
+            ::Error("TSimpleAnalysis::Analyze", "Multiple trees inside %s", fInputFiles[0].c_str());
+            return "";
+         }
+      }
+   }
+   // If treeName is yet empty, error occours
+   if (treeName.empty()) {
+      ::Error("TSimpleAnalysis::Analyze", "No tree inside %s", fInputFiles[0].c_str());
+      return "";
+   }
+   return treeName;
+}
+
+////////////////////////////////////////////////////////////////////////////////
+/// Execute all the TChain::Draw() as configured and stores the output histograms.
+/// Returns true if the input arguments create correctly the output file.
+
+bool TSimpleAnalysis::Analyze()
+{
+   // Disambiguate tree name from first input file:
+   // just try to open it, if that works it's an input file.
+   int oldLevel = gErrorIgnoreLevel;
```

Comment:
Add // silence possible error message from TFile constructor if this is a tree name.